foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/mutating/webhook/ConfigMutatingWebhook.java

```diff
@@ -33,31 +48,124 @@
+                JsonNode patch = JsonDiff.asJson(source, mapper.readTree(target.toString()));
+                response = new AdmissionResponseBuilder()
+                        .withAllowed(true)
+                        .withUid(request.getUid())
+                        .withPatch(Base64.getEncoder().encodeToString(patch.toString().getBytes()))
+                        .withPatchType("JSONPatch")
+                        .build();
+                logger.info("The kubernetes resource request has been updated.");
+            } else {
+                response = new AdmissionResponseBuilder()
+                        .withAllowed(true)
+                        .withUid(request.getUid())
+                        .build();
+                logger.info("No updates have made to the kubernetes resource request.");
+            }
+
+        } catch (Exception e) {
+            logger.error("Error injecting the config store value", e);
+            // create failed response
+            response = new AdmissionResponseBuilder()
+                    .withAllowed(false)
+                    .withUid(request.getUid())
+                    .withStatus(new StatusBuilder().withStatus("Failed").build())
+                    .build();
+        }
+
+        return new AdmissionReviewBuilder()
+                .withResponse(response)
+                .withApiVersion(admissionReviewRequest.getApiVersion())
+                .build();
+    }
+
+    public String getProperty(String groupName,String propertyName) {
+        PropertyKey propertyKey = new PropertyKey(groupName, propertyName);
+        Property property;
+        try {
+            property = this.configLoader.read(propertyKey);
+            return property.getValue();
+        } catch (Exception e) {
+            logger.error(String.format("Error reading property - groupName: %s, propertyName: %s", groupName, propertyName), e);
+        }
+        return null;
+    }
+
+    private StringBuilder replaceConfigStoreKeyWithConfigValue(String source) {
+        StringBuilder target = new StringBuilder();
+        String[] data = source.split(CONFIG_STORE_INJECT_START);
+        // if the config store key is found
+        if (data.length > 1) {
+            target.append(data[0]);
+            logger.info("Config store properties injection requested: ");
+            for (var i=1; i<data.length; i++) {
+                int contentIndex = data[i].indexOf(CONFIG_STORE_INJECT_END);
+                String[] key = data[i].substring(0, contentIndex).split(PROPERTY_KEY_DELIMITER);
+                Map<String, String> keyMap = convertToKeyMap(key);
+                String value = getProperty(keyMap.get(GROUP_NAME), keyMap.get(PROPERTY_NAME));
+                target.append(value);
+                if (data[i].length() > contentIndex + 1) {
+                    target.append(data[i].substring(contentIndex+1));
+                }
+            }
+        }
+
+        return target;
```

Comment:
A: Based on how you use this return value, seems cleaner to return the String instead of StringBuilder.  Also, I feel like the name could be a little better. Maybe something like `updatedJson` or `processedBody`.